src/ensembl/production/metadata/api/search/search.py

```diff
@@ -0,0 +1,342 @@
+#   regarding copyright ownership.
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from typing import Optional, List
+
+from pydantic import BaseModel
+from sqlalchemy.orm import Session, joinedload
+
+from ensembl.production.metadata.api.models import (
+    Genome, Dataset, DatasetAttribute,
+    EnsemblRelease, GenomeRelease, GenomeDataset, Attribute,
+    ReleaseStatus
+)
+
+
+# ============================================================================
+# PYDANTIC SCHEMAS
+# ============================================================================
+
+class GenomeSearchDocument(BaseModel):
+    """Schema for genome search indexing"""
+
+    # Direct fields from Genome/Organism/Assembly
+    genome_uuid: str
+    common_name: Optional[str] = None
```

Comment:
As a comment, if you set minimum Python version to 3.10, you can do the following (I find it more readable and removes the import of `Optional`):
```suggestion
    common_name: str | None = None
```